que.go

```diff
@@ -216,46 +216,56 @@ func (c *Client) LockJob(queue string) (*Job, error) {
 	}
 
 	j := Job{pool: c.pool, conn: conn}
-	err = conn.QueryRow("que_lock_job", queue).Scan(
-		&j.Queue,
-		&j.Priority,
-		&j.RunAt,
-		&j.ID,
-		&j.Type,
-		&j.Args,
-		&j.ErrorCount,
-	)
-	if err != nil {
-		c.pool.Release(conn)
-		if err == pgx.ErrNoRows {
-			return nil, nil
+
+	for {
```

Comment:
Under high concurrency this race condition isn't *that* rare.  In the tests I ran today, with four workers pulling from a queue, it happened 50,000 times for a queue of 75,000 items.  Admittedly, the workers weren't actually working so they were more likely to hit the race than a normal application probably is.

Perhaps a value of 10 would have workers getting really unlucky under high concurrency.  And like you said, we could export the error, so people who are knowingly in this situation could avoid problems by checking for it.